vnext/ReactUWP/Threading/HeadlessJSMessageQueueThread.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <ReactWindowsCore/BatchingMessageQueueThread.h>
+#include <cxxreact/MessageQueueThread.h>
+
+namespace react {
+namespace uwp {
+
+/// This is a Message Queue Thread that wraps two seperate Message Queue
+/// Threads. The idea is to allow starting a react instance before a Window or
+/// UI Dispatcher are available (running background tasks) Then once
+/// AttachRootView is called a new Message Queue Thread can be provided with
+/// setUIMessageQueue.
+
+class HeadlessJSMessageQueueThread
```

Comment:
for transparency, this class is why I introduced a preprocessor. imo this is the weirdest part of this change.
We can't change what Thread Queue we already passed to the module descriptors so this is a proxy allowing us to change what we proxy to.